plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java

```diff
@@ -58,10 +64,11 @@ public MethodVisitor visitMethod(int access, String name, String descriptor, Str
     public void visitEnd() {
         spockMethodVisitor.getTestMethodPatterns().stream()
             .filter(methodPattern -> {
-                String methodPatternRegex = escapeRegEx(methodPattern).replaceAll("#\\w+", "\\\\w+");
+                //detects a valid spock parameter and replace it with a wildcards http://spockframework.org/spock/docs/1.3/data_driven_testing.html#_more_on_unrolled_method_names
+                String methodPatternRegex = escapeRegEx(methodPattern.replaceAll(SPOCK_PARAM_PATTERN, PARAM_PLACEHOLDER)).replaceAll(PARAM_PLACEHOLDER, WILDCARD) + WILDCARD_SUFFIX;
```

Comment:
I'd extract the `methodPattern.replaceAll(SPOCK_PARAM_PATTERN, PARAM_PLACEHOLDER)` part to a `normalizeMethodName()` method to make it more readable.